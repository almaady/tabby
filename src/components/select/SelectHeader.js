import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import ButtonIcon from "../buttons/ButtonIcon";
import Icon from "../icons/Icon";

/**
 * @class SelectHeader - It is the title of the select
 */
class SelectHeader extends React.PureComponent {
  /**
   * @property {string} icon - The icon of the Header
   * @property {string} title - The title of the header
   * @property {string} className - ClassName if you need to customize
   * @property {func} onClick - The function to open the Select List
   * @property {bool} listOpen - Determines if the Select List is visible
   */
  static propTypes = {
    icon: PropTypes.string,
    title: PropTypes.string.isRequired,
    className: PropTypes.string,
    onClick: PropTypes.func.isRequired,
    listOpen: PropTypes.bool
  };

  render() {
    const { className, icon, title, onClick, listOpen } = this.props;

    return ( 
      <div className={classNames('select__header', className)} onClick={onClick}>
        {icon && <Icon icon={icon} color="not-gray-l2" size="R" />}
        <p>{title}</p>
        <ButtonIcon icon={listOpen ? "top" : "down"} color="not-gray-l1" size="XXS" onClick={()=>{}}/>
      </div>
    );
  }
}

SelectHeader.defaultProps = {
  icon: "",
  className: "",
  listOpen: false
};

export default SelectHeader