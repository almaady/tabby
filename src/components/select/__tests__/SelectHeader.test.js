import React from "react";
import {shallow, mount} from "enzyme";
import sinon from "sinon"

import SelectHeader from "../SelectHeader";

describe("SelectHeader", () =>{
  const toggle = sinon.stub();
  let props={
    title: "Title",
    onClick: toggle
  }
  
  it("should render correctly with default state", () => {
    const component = shallow(<SelectHeader {...props} />);
    expect(component).toMatchSnapshot();
  });
  it("should call toggle action", () => {
    const component = shallow(<SelectHeader {...props} />);
    component.simulate('click');
    expect(toggle.called).toBeTruthy();
  });
  it("should change icon", () => {
    const component = mount(<SelectHeader {...props} listOpen={true} />);
    expect(component.find('ButtonIcon').props().icon).toEqual('top');
  });
});