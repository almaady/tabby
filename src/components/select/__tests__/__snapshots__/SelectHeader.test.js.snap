// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectHeader should render correctly with default state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectHeader
    className=""
    icon=""
    listOpen={false}
    onClick={[Function]}
    title="Title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "",
        <p>
          Title
        </p>,
        <ButtonIcon
          borderColor=""
          className=""
          color="not-gray-l1"
          disabled={false}
          icon="down"
          onClick={[Function]}
          size="XXS"
          type="button"
        />,
      ],
      "className": "select__header",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Title",
        },
        "ref": null,
        "rendered": "Title",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "borderColor": "",
          "className": "",
          "color": "not-gray-l1",
          "disabled": false,
          "icon": "down",
          "onClick": [Function],
          "size": "XXS",
          "type": "button",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "",
          <p>
            Title
          </p>,
          <ButtonIcon
            borderColor=""
            className=""
            color="not-gray-l1"
            disabled={false}
            icon="down"
            onClick={[Function]}
            size="XXS"
            type="button"
          />,
        ],
        "className": "select__header",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Title",
          },
          "ref": null,
          "rendered": "Title",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "borderColor": "",
            "className": "",
            "color": "not-gray-l1",
            "disabled": false,
            "icon": "down",
            "onClick": [Function],
            "size": "XXS",
            "type": "button",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
